### ISSUE TITLE
Automate A11y Testing

### ISSUE THREAD

**Issue by Garbee** [google/material-design-lite#1892]
With 2.x along with expanding out existing coverage, we should integrate a11y testing. This can be done with tools such as [axe-core](https://github.com/dequelabs/axe-core). I think we should have a11y coverage for each component individually, then also build one or two example pages to test as well, to cover how the components work together as a whole.

**Comment by addyosmani** [comment]
Good idea!

We maintain both https://github.com/GoogleChrome/accessibility-developer-tools and https://github.com/addyosmani/a11y (uses the former and is targeted at CI). Is there something axe-core gives us that might not be covered by these other tools? The former in particular is quite comprehensive and powers the Accessibility DevTools extension too.

**Comment by Garbee** [comment]
I was looking into things that don't use or depend on the Chrome A11y Devtools plugin since it is being slowly wrapped into core. It eventually would not be kept up-to-date or bugfixed/improved.

Also, talking things over at the Dev Summit, we are going to end up with a light threshold on a11y things. We could probably do the bulk of the testing directly with mocha instead of even needing an extra package. Since with 2.x we are moving away from creating elements through the JS, that means our a11y will mostly end up in docs to aid developers in properly implementing components.

**Comment by Garbee** [comment]
Dropped the V2 milestone on this. While it would be super-nice to figure out a11y before the major lands, we just don't have the known capacity to handle it. Let's instead focus on our bigger infrastructure changes and get them hammered out right. We can think about a11y when building components from the start, and integrate testing it in later releases.
